micro-rdk/src/common/robot.rs

```diff
@@ -217,17 +241,26 @@ impl LocalRobot {
     // component configs within the response are consumed and the corresponding components are generated
     // and added to the created robot.
     pub fn from_cloud_config(
+        exec: Executor,
+        part_id: String,
         config_resp: &ConfigResponse,
         registry: Box<ComponentRegistry>,
         build_time: Option<DateTime<FixedOffset>>,
     ) -> Result<Self, RobotError> {
         let mut robot = LocalRobot {
+            executor: exec,
+            part_id,
             resources: ResourceMap::new(),
             // Use date time pulled off gRPC header as the `build_time` returned in the status of
             // every resource as `last_reconfigured`.
             build_time,
+
             #[cfg(feature = "data")]
             data_collector_configs: vec![],
+            #[cfg(feature = "data")]
+            data_manager_sync_task: None,
+            #[cfg(feature = "data")]
```

Comment:
this isn't necessary since the field isn't feature-gated anywhere else and is referenced in `Drop`

Reply:
Good catch, I will fix that.